performance/include/traccc/performance/timer.hpp

```diff
@@ -0,0 +1,46 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/performance/timing_info.hpp"
+
+// System include(s).
+#include <chrono>
+#include <string_view>
+
+namespace traccc::performance {
+
+/// Object used for measuring execution time.
+/// Start time measured at construction. End time at destruction.
+/// Creating more than one start & stop of timer with the same timer_name &
+/// sharing timing info will lead to incrementing the total time for that name
+///
+class timer {
+
+    public:
+    /// Start time measurement
+    /// @param timer_name name to be printed out identifying what is measured
+    /// @param t_info shared_ptr to timing_info where to store timings
+    timer(const std::string_view timer_name, timing_info* t_info);
+
+    /// End time measurement
+    ~timer();
+
+    private:
+    /// Start time (measured at construct time)
+    std::chrono::high_resolution_clock::time_point m_start;
+
+    /// Name of measurement
+    std::string_view m_name;
```

Comment:
So... [std::string_view](https://en.cppreference.com/w/cpp/string/basic_string_view) is an excellent function argument. It allows calls with `const char*` literals to have a zero overhead. Let's say when you instantiate the class with:

```c++
traccc::performance::timer t{"Foo", bar};
```

But as the name implies, this type is merely a view of a string whose ownership is managed somewhere else. It is **very** similar to the VecMem view types in this sense.

So to make `traccc::performance::timer` safe, its member variable should actually be a simple old [std::string](https://en.cppreference.com/w/cpp/string/basic_string) instead. So that in case it's created using a temporary string, it would still work well. Let's say in the following messed up situation:

```c++
traccc::performance::timer t{"Foo"s, bar};
```

See the notes towards the bottom of: https://en.cppreference.com/w/cpp/string/basic_string_view

So:
  - Constructor parameter: `std::string_view`
  - Member variable: `std::string`